<?xml version="1.0"?>
<launch>

	<param name="use_sim_time" value="true" />
    
    <node pkg="rosbag" type="play" name="playbag" output="screen"
          args="--pause --clock --rate 1
                $(find lidar_eskf)/dat/NSH1109_circle/lidar.bag
                $(find lidar_eskf)/dat/NSH1109_circle/imu.bag
                $(find lidar_eskf)/dat/NSH1109_circle/camera.bag"/>
                
	<node pkg="tf" type="static_transform_publisher" name="world_view" args="0 0 0 0 0 3.1415926 world world_view 100"/> 
    <node pkg="tf" type="static_transform_publisher" name="imu_camera" args="0 0 0 1.5707963 0 -1.5707963 imu camera 100"/>

	<node pkg="lidar_eskf" type="lidar_eskf_node" name="lidar_eskf" output="screen" >
	
	    <remap from="/lidar_eskf/cloud"        to="/3d_lidar_cloud"/>
        <remap from="/imu"                     to="/imu_driver/imu"/>
	    <remap from="/meas"                    to="/lidar_eskf/posterior"/>
	  
	    <param name="robot_frame"              value="imu"/>
	    <param name="imu_frame"                value="imu"/>
		<param name="imu_enabled"              value="true"/>
		<param name="imu_has_quat"             value="true"/>
		<param name="imu_transform"            value="false"/>
        <param name="map_file_name"            value="$(find lidar_eskf)/map/nsh_1109.bt"/>
        <param name="octree_resolution"        value="0.05"/>
        <param name="max_obstacle_dist"        value="0.5"/>
        <param name="ray_sigma"                value="1.0"/>
        <param name="cloud_resolution"         value="0.1"/>
		<param name="laser_type"               value="pointcloud"/>
        <param name="cloud_range"              value="30.0"/>
        <param name="set_size"                 value="500"/>
        <param name="pcd_file"                 value="$(find lidar_eskf)/dat/recmap_nsh_1109.pcd"/>
        
        <param name="imu_frequency"             value="200"/>
        <param name="sigma_acceleration"        value="4.0"/>
        <param name="sigma_gyroscope"           value="0.1"/>
        <param name="sigma_acceleration_bias"   value="0.002"/>
        <param name="sigma_gyroscope_bias"      value="0.001"/>
        <param name="gravity"                   value="9.82"/>
        
        <param name="init_bias_acc_x"           value="0.0"/> #-0.05
        <param name="init_bias_acc_y"           value="0.0"/> #0.025
        <param name="init_bias_acc_z"           value="0.0"/> #-0.032
        
        <param name="acc_queue_size"            value="5"/>

	</node>
	
	<node pkg="octomap_server" type="octomap_server_node" name="eskf_octomap_server" 
	      args="$(find lidar_eskf)/map/nsh_1109.bt">
	      <param name="frame_id"           value="world"/>
	      <param name="base_frame_id"      value="world_view"/>
	</node>
	
	<node pkg="rviz" type="rviz" name="gpf_test_rviz" required="false" args="-d $(find lidar_eskf)/rviz/lidar_eskf.rviz"/>
</launch>
